python/pyspark/sql/dataframe.py

```diff
@@ -1181,18 +1181,33 @@ def approxQuantile(self, col, probabilities, relativeError):
         Space-efficient Online Computation of Quantile Summaries]]
         by Greenwald and Khanna.
 
-        :param col: the name of the numerical column
+        :param col: the name of the numerical column, or a list/tuple of
+          numerical columns.
         :param probabilities: a list of quantile probabilities
           Each number must belong to [0, 1].
           For example 0 is the minimum, 0.5 is the median, 1 is the maximum.
         :param relativeError:  The relative target precision to achieve
           (>= 0). If set to zero, the exact quantiles are computed, which
           could be very expensive. Note that values greater than 1 are
           accepted but give the same result as 1.
-        :return:  the approximate quantiles at the given probabilities
+        :return:  the approximate quantiles at the given probabilities. If
+          the input `col` is a string, the output is a list of float. If the
+          input `col` is a list or tuple of strings, the output is also a
+          list, but each element in it is a list of float, i.e., the output
+          is a list of list of float.
         """
-        if not isinstance(col, str):
-            raise ValueError("col should be a string.")
+        if not isinstance(col, (str, list, tuple)):
+            raise ValueError("col should be a string, list or tuple.")
+
+        isStr = isinstance(col, str)
```

Comment:
That's true, but if we can get rid of one of the unused private methods on the Scala side I'm all for that. Can we not simply return the first element of the result if it is length 1, otherwise return the result?